### PR TITLE
docs: Add Contributing Guidelines

### PR DESCRIPTION
**Pull request by vvatanabe**
**Describe the change**

- Contributing Guidelines have been added. For details, refer to CONTRIBUTING.md.
- A link to CONTRIBUTING.md has been added in the README.md.
- The Integration Test section from README.md has been moved to CONTRIBUTING.md.
- The outlines for Installation and Usage in README.md have been aligned.

**Tests**
None

**Additional context**
None

### CODE REVIEW

**Comment by vvatanabe** [comment]
@sashabaranov I've written the Contributing Guidelines. Please check them. If there are any rules you'd like to add, remove, or update, please let me know.

**Comment by vvatanabe** [comment]
@sashabaranov It looks like you're busy, so I'll go ahead and merge this for now. If there are any issues or feedback regarding this guideline, I'll respond right away.

**Comment by vvatanabe** [comment]
Oh, I see. I can't approve a pull request that I created myself.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+### Code Style
+In this project, we adhere to the standard coding style of Go. Your code should maintain consistency with the rest of the codebase. To achieve this, please format your code using tools like `goimports` and resolve any syntax or style issues with `golangci-lint`.
+
+**Run goimports:**
+```
+go install golang.org/x/tools/cmd/goimports@latest
+```
+
+```
+goimports -w .
```
Comment (sashabaranov) [inline CONTRIBUTING.md:53]: This is a really great PR, thank you so much @vvatanabe! Sorry for such a long delay with the review.

Just a note: it seems that golangci-lint does include goimports into itself, so it's enough to run to just run the golangci-lint.

To reproduce:

```diff
diff --git a/completion_test.go b/completion_test.go
index 844ef48..3a63504 100644
--- a/completion_test.go
+++ b/completion_test.go
@@ -4,6 +4,7 @@ import (
        . "github.com/sashabaranov/go-openai"
        "github.com/sashabaranov/go-openai/internal/test/checks"

+       "testing"
        "context"
        "encoding/json"
        "errors"
@@ -12,7 +13,6 @@ import (
        "net/http"
        "strconv"
        "strings"
-       "testing"
        "time"
 )
```

Produces:

```
% golangci-lint run ./...
completion_test.go:7: File is not `goimports`-ed (goimports)
	"testing"
```

**Review by sashabaranov — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (vvatanabe) [reply CONTRIBUTING.md:53]: @sashabaranov Thanks for the feedback. Indeed, `golangci-lint` does give warnings, but to fix them, you need to run goimports yourself. I've added instructions on how to run `goimports` to help developers who are not familiar with Go handle it smoothly.

---
Don't worry about the delayed response. You should focus on what you need to prioritize! It's more important.

**Pull request merged**